cookiecutter_project_upgrader/cli.py

```diff
@@ -45,3 +54,29 @@ def _load_context(context_file: str):
         exit(1)
     context = json.loads(context_str)
     return context
+
+
+def _is_valid_file(zip_file: str):
+    # we check only for local files for valid, for remote URLs, cookiecutter can do the test if its valid.
+    REPO_REGEX = re.compile(
+        r"""
+        ((((git|hg)\+)?(git|ssh|file|https?):(//)?)
+        |
+        (\w+@[\w\.]+)
+        )
+        """,
+        re.VERBOSE,
+        )
+    try:
+        if not bool(REPO_REGEX.match(zip_file)):
+            if ZipFile(Path(zip_file)).testzip() is None:
+                return True
+    except BadZipFile:
+        click.echo(f"Bad Zip file at: {zip_file}\n"
+                   f"Make sure your zip file is correct.")
+        exit(1)
+    except FileNotFoundError:
+        click.echo(f"Zip file not found at: {zip_file}\n"
+                   f"Make sure your path is correct.")
+        exit(1)
```

Comment:
nit: click [supports](https://click.palletsprojects.com/en/7.x/api/#click.echo) echoing to stderr with `err=True`

also realised that with my suggestion above, this error won't be raised, for which we can depend on the upstream error from cookiecutter itself:

```suggestion
    except BadZipFile:
        click.echo(f"Bad Zip file at: {zip_file}\n"
                   f"Make sure your zip file is correct.", err=True)
        exit(1)
```